### DISCUSSION TITLE
How can i check which terminal i am at ?

### DISCUSSION THREAD

**Discussion by almog-hindi** [community/community#135579] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Show & Tell

### Body

I got to check which terminal i used so i can know if the terminal supports emojis or not, how can i do it ?

**Comment by almoghindi** [comment, marked as answer]
import { execSync } from "child_process";

function getParentProcessName() {
  try {
    const processId = process.ppid;
    const command = `wmic process where processid=${processId} get Caption`;
    const output = execSync(command).toString().trim().split("\n");
    return output[1] ? output[1].trim() : "Unknown";
  } catch (err) {
    console.error("Error determining parent process name:", err);
    return "Unknown";
  }
}

export function determineTerminal() {
  const parentProcessName = getParentProcessName();

   if (parentProcessName.includes("cmd.exe")) {
     return "Command Prompt (CMD)";
   } else if (parentProcessName.includes("powershell.exe")) {
     return "PowerShell";
   } else if (parentProcessName.includes("wt.exe")) {
     return "Windows Terminal";
   } else if (parentProcessName.includes("iTerm.app")) {
     return "iTerm2 (macOS)";
   } else if (parentProcessName.includes("Terminal.app")) {
     return "Terminal.app (macOS)";
   } else if (parentProcessName.includes("gnome-terminal")) {
     return "GNOME Terminal (Linux)";
   } else if (parentProcessName.includes("konsole")) {
     return "Konsole (Linux)";
   } else if (parentProcessName.includes("alacritty")) {
     return "Alacritty";
   } else if (parentProcessName.includes("hyper")) {
     return "Hyper";
   } else if (parentProcessName.includes("xterm")) {
     return "xterm";
   } else if (parentProcessName.includes("urxvt")) {
     return "rxvt or rxvt-unicode";
   } else if (parentProcessName.includes("terminator")) {
     return "Terminator";
   } else if (parentProcessName.includes("tmux")) {
     return "tmux";
   } else if (parentProcessName.includes("bash")) {
     return "Bash";
   } else {
     return "Unknown or other terminal emulator";
   }
}
